src/components/WwwFrame/DisclaimersContentful.vue

```diff
@@ -0,0 +1,131 @@
+		};
+	},
+	inject: ['apollo', 'cookieStore'],
+	apollo: {
+		query: bannerQuery,
+		preFetch: true,
+		result({ data }) {
+			this.disclaimerContent = [];
+			// gather contentful content and the uiSetting key ui-global-promo
+			const contentfulContent = data?.contentful?.entries?.items ?? [];
+			const uiGlobalPromoSetting = contentfulContent.find(item => item.fields.key === 'ui-global-promo');
+
+			// exit if missing setting or fields
+			if (!uiGlobalPromoSetting || !uiGlobalPromoSetting.fields) {
+				return false;
+			}
+			// uiGlobalPromoSetting can contain an array of different banners with
+			// different start/end dates first determine if setting is enabled.
+			const isGlobalSettingEnabled = settingEnabled(
+				uiGlobalPromoSetting.fields,
+				'active',
+				'startDate',
+				'endDate'
+			);
+
+			// if setting is enabled determine which banner to display
+			if (isGlobalSettingEnabled) {
+				const activePromoBanner = uiGlobalPromoSetting.fields.content.find(promoContent => {
+					return settingEnabled(
+						promoContent.fields,
+						'active',
+						'startDate',
+						'endDate'
+					);
+				});
+
+				// gather all inactive promo banners by their start and end dates
+				const inactivePromoBanners = uiGlobalPromoSetting.fields.content.filter(promoContent => {
+					const hiddenUrls = promoContent?.fields?.hiddenUrls ?? [];
+					// check hiddenUrl for display of disclaimers
+					if (hiddenUrls.includes(this.$route.path)) {
+						return false;
+					}
+					if (promoContent.fields.active) {
+						return false;
+					}
+					return settingWithinDateRange(
+						promoContent.fields,
+						'startDate',
+						'endDate'
+					);
+				});
+
+				if (activePromoBanner) {
+					// check for visibility based on current route and hiddenUrls field
+					const hiddenUrls = _get(activePromoBanner, 'fields.hiddenUrls', []);
+					if (hiddenUrls.includes(this.$route.path)) {
+						return false;
+					}
+
+					// check for visibility on promo session override
+					const showForPromo = _get(activePromoBanner, 'fields.showForPromo', false);
+					if (this.hasPromoSession && !showForPromo) {
+						return false;
+					}
+
+					// set the disclaimer text if it exists in active promo banner
+					const activeDisclaimerText = activePromoBanner?.fields?.disclaimers ?? null;
+
+					// if there's an active promo banner with a disclaimer,
+					// push that disclaimer to the disclaimerContent for display
+					if (activeDisclaimerText) {
+						this.disclaimerContent.push(documentToHtmlString(activeDisclaimerText));
+					}
+				}
+
+				// go through the inactive promoBanners, if within date range and disclaimer text exists
+				// push that disclaimer text to disclaimerContent
+				if (inactivePromoBanners.length > 0) {
+					inactivePromoBanners.forEach(item => {
+						const itemDisclaimer = item?.fields?.disclaimers ?? null;
+						if (itemDisclaimer) {
+							this.disclaimerContent.push(documentToHtmlString(itemDisclaimer));
```

Comment:
Does it matter if the list number that is used for a given disclaimer changes over time? With this forEach the numbers could change as different promos end. If they need to be consistent for the entire duration of the promo, then the list number might need to be defined in Contentful. I could see that getting really complicated though, so I hope that the number doesn't matter for the extra promos.

Reply:
I think we're ok without numeric assignment at this time as only an "active" promo could be visible at the top with a jump link requiring any correlation. The "active" promo is extracted first and filtered out of the "inactive" set so it will also always be 1st in the list to match that `1` link at the top.

Let's keep an eye on this as it gets used though as we may have to get more sophisticated with tracking disclaimers along any presented content...